### PR TITLE
Comment riscv_set_register, register_write_direct

### LINKED ISSUE #241: Comment on difference between riscv_set_register and register_write_direct

### PR DESCRIPTION
**Pull request by timsifive**
Fixes #241

Change-Id: Ia199f15106a0bda465d3918d052ddd4d03655031

### CODE REVIEW

File: src/target/riscv/riscv.c [hunk 1]
```diff
@@ -1763,6 +1763,10 @@ bool riscv_has_register(struct target *target, int hartid, int regid)
 	return 1;
 }
 
+/**
+ * This function is called when the debug user wants to change the value of a
+ * register. The new value may be cached, and may not be written until the hart
+ * is running again. */
```
Comment (mwachs5) [inline src/target/riscv/riscv.c]: "is resumed"?

**Review by mwachs5 — DISMISSED** [review]
Pretty good, but is one preferable to the other in the general case? If you call register_write_direct does it invalidate the cache?

**Comment by timsifive** [comment]
There is no general case.

Either you're passing on a user/higher level request, in which case you're using riscv_set_register(). Otherwise when the register writing is done for reasons internal to the RISC-V debug support you're using register_write_direct()

**Comment by timsifive** [comment]
Could I get a review for this?

**Review by palmer-dabbelt — APPROVED** [review]

**Pull request merged**